lib/libzrepl/data_conn.c

```diff
@@ -377,6 +388,18 @@ uzfs_zvol_worker(void *arg)
 		zio_cmd_free(&zio_cmd);
 		goto drop_refcount;
 	}
+
+	if (rebuild_cmd_req && (hdr->opcode == ZVOL_OPCODE_READ) &&
+	    (data_err == EIO) && (!md_err)) {
```

Comment:
EIO? It should be ENOENT right..?

Reply:
Yes.
It should be `ENOENT`.
`dmu_buf_hold_array_by_dnode` returns `EIO` if `dbuf_hold_impl` returns db=NULL (dh->dh_err will be set to ENOENT in `dbuf_hold_impl`).
This is check is removed as per new implementation.